optuna/distributions.py

```diff
@@ -658,18 +658,8 @@ def json_to_distribution(json_str: str) -> BaseDistribution:
             log = json_dict.get("log")
```

Comment:
Sorry to jump in - I think `log` can be `None` if the given json_dict doesn't have the key. In such case, shall we assign `log = False` in this function? 
I'm planning to do the same for `IntDistribution`'s `log` and `step` if it sounds fine with you.

Reply:
Sounds good!

Reply:
PTAL at 607f70eb00efa4aece5e25b52f08ada13e35d97c. I've specified default value for `log` argument, since it's used by both float and int distributions. You can omit it and focus on `step` as i assume you'll rebase at some point when this PR is merged.

Reply:
Thank you so much for prompt response, that sounds even better! Much appreciated.